lib/puppet/network/authorization.rb

```diff
@@ -1,7 +1,17 @@
 module Puppet::Network
   module Authorization
-    def self.authconfigloader_class=(_)
-      # legacy auth is not supported, ignore
+    class << self
+      def authconfigloader_class=(klass)
+        @authconfigloader_class = klass
+      end
+
+      # Verify puppetserver registered its authconfigloader class
+      def check_authorization(method, path)
```

Comment:
I think that leaving the `authconfigloader_class=` setter for backwards compatibility during an upgrade is good, but providing a more explicit name going forward might be helpful. Like `external_auth_configured?`

Reply:
I originally had a predicate like that, but then the method either returned true or raised, so it wasn't really a predicate. So I was thinking `check_authorization` made that more explicit.

Reply:
maybe `check_external_authorization`?

Reply:
I like that! Could we also maybe leave `authconfigloader_class=` but add a new method that sets a boolean? I'm mainly concerned about giving a thing a class that's called a loader, but only ever checked for existence.

Reply:
> but add a new method that sets a boolean?

Sorry I didn't follow, were you thinking `authconfigloader_class=` would call that other method?

Reply:
Yeah I was thinking like:
```rb
# deprecated api, remove in Puppet 8
def authconfigloader_class=(_)
  acknowledge_external_auth!
end

def acknowledge_external_auth!
  @external_auth = true
end
...
  unless @external_auth
    ....
```
I was thinking then in Server 7 we could get rid of our null auth implementation. But that doesn't jive with the Server 7 & Agent 6 upgrade path. We'll have to wait until Server 8 to remove our dummy auth impl, and at that point we could do this dance. So 🤷

Reply:
If puppet code was packaged into puppetserver, then we could get rid of the null auth and this issue could be avoided altogether. But yeah I think we're stuck with `authconfigloader_class=` until 8.